ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -47,19 +49,37 @@ var (
 	// headless svc will be of the form: {kuberay-cluster-name}-headless-worker-svc
 	headlessServiceSuffix = "headless-worker-svc"
 
+	// k8s Pod informer to query current cluster Pod list
+	// TODO: refactor webhook to remove global vars
+	podLister listersv1.PodLister
+
 	// map of pod slices to workers in the slice
-	sliceToWorkers map[slice][]worker
+	sliceToWorkerIDs map[slice][]int
+
+	// mutex lock for webhook go routines accessing Informer cache
+	cacheMutex sync.Mutex
 
-	// map of pod slices to TPU_WORKER_HOSTNAMES in that pod slice
-	sliceToHostnames map[slice]string
+	// mutex lock for webhook go routines accessing sliceToWorkerIDs mapping
+	readMutex sync.Mutex
```

Comment:
You don't need a separate read and write mutex. You can use a single `RWMutex` and call `Rlock()` for reads and `Lock()` for writes

Reply:
Switched to using a single `RWMutex` in [d32a864](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/d32a864b8ce2baad2566893ab0b79eea486ee00c).